redis-tools

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+#
+# This script provides tools to be used with the TurbiniaEvidence keys in
+# Redis. There are 5 main functions: keys, values, delete, dump and restore.
+# The first five will query over Redis to find keys that have the same value
+# in a specific field, as determined by the passed arguments. Restore will
+# restore all keys dumped in a directory. If [Field Name] and [Field Value] 
+# are replaced with "all", all keys will be selected. 
+# 
+# ./redis-tools keys [Field Name] [Field Value]
+# The matching keys are printed out.
+# 
+# ./redis-tools values [Field Name] [Field Value]
+# The values of the matching keys are printed out.
+# 
+# ./redis-tools delete [Field Name] [Field Value]
+# The matching keys are deleted from redis.
+# 
+# ./redis-tools dump [Field Name] [Field Value] [Directory to Dump]
+# The matching keys are dumped to the indicated directory.
+# 
+# ./redis-tools restore [Directory to Restore] 
+# The keys dumped in the directory are restored.
+#
+# Examples: 
+# ./redis-tools keys all
+# ./redis-tools values id f5bd194d9424427c9d82a57eedc531d0
+# ./redis-tools dump request_id 09879dbc90a44c6db80dc9f68113d829 storagefolder
+# ./redis-tools delete all
+# ./redis-tools restore storagefolder
+
+action=$1 # "keys" / "values" / "delete" / "dump" / "restore"
+arg1=$2
+arg2=$3
+arg3=$4
+
+if [[ "$action" == "restore" ]]
+then
+  for filename in "$arg1"/*; do
+    key="$(echo "$filename" | sed 's|.*/||')"
+    echo Restoring "$key"
+    cat "$filename" | redis-cli -x restore "$key" 0
+  done
+else
+  # Ignores [Field Value] if "all" is passed
+  if [[ "$arg1" == "all" ]]; then arg3=$arg2; fi
+
+  if [[ "$action" != "values" ]]; then
+    declare -a key_array
+    if [[ "$action" == keys ]] || [[ "$arg1" != "all" ]]; then echo Keys found:; fi
+  fi
+
+  for key in $(redis-cli --scan); do 
+    key_type=${key%:*}
```

Comment:
This syntax is not super uncommon, could you potentially add a comment for what this does?

Reply:
Sure, added a comment to explain it.